versions/1/integration/data_stream/manifest.spec.yml

```diff
@@ -212,5 +212,10 @@ spec:
               type: array
               items:
                 type: string
+            allow_routing:
```

Comment:
If I remember correctly, so far this structure was built after the structure that Elasticsearch has. The advantage with this is that Fleet can just pass it on to fleet-server and it will just work. If this is the case, we should not allow_routing in here. Instead I think there is a `names` flag from Elasticsearch which could be used here? https://www.elastic.co/guide/en/elasticsearch/reference/current/defining-roles.html#roles-indices-priv

Assuming `names` works, it would make the implementation a bit different in Fleet and the spec. The spec would have to validate that it does not go beyond the permissions the fleet-server has. On the spec side, it should be validated that it is inside the data stream naming scheme.

Reply:
I think I prefer the semantics that this PR proposes over letting an integration directly configure `names`. If we retain the semantics, I agree that it should move out of the `elasticsearch.privileges`.

The reason I prefer `allow_routing` is that it only allows elevated privileges on data streams that conform to the `type` that the integration has specified. This ensures the integration only has the permissions it needs. A logging integration will never need privileges to `metrics-*-*`, for example.  

Therefore, I propose to just add the `allow_routing` flag to the top level:

```yaml
type: logs
allow_routing: true
```

This would result in privileges for `logs-*-*`.

Reply:
> data streams that conform to the type that the integration has specified

This might not always be the case. For example currently firehose only supports logs but soon is adding metrics. I'm not sure if it comes out the same stream or a separate setup is needed (still to be figured out). If it comes out of the same stream, it will need routing to logs and metrics.

The reason I like `names` is because we don't have to introduce a new config option just yet and wait on a few "routing integrations" to happen and then potentially introduce a specific config option for these that is more restrictive and descriptive. I'm worried we take one example and abstract based on it.

Reply:
>  If it comes out of the same stream, it will need routing to logs and metrics.

Currently, the [`data_stream_router`](https://github.com/elastic/elasticsearch/pull/76511) can't change the `type` of the data stream to avoid that an event is accidentally or maliciously routed from `logs-router-default` to a non-logs data stream.

If that's a requirement for firehose or others, we'll also need to change that.

Reply:
Lets consider it a feature that it is not possible at the moment and add it only when needed.